### ISSUE TITLE
stucked using single-machine multi-gpus training

### ISSUE THREAD

**Issue by Number1JT** [open-mmlab/OpenPCDet#153]
training using command in tutorial，the utility rate stay at 0 for a very long time，It looks like the training is  totally stuck. 
Any suggestion to slove this problem？THX

> CUDA_VISIBLE_DEVICES=4,5,6,7 OMP_NUM_THREADS=1 python -m torch.distributed.launch --nproc_per_node=4 train.py --launcher pytorch --cfg_file  cfgs/kitti_models/pv_rcnn.yaml --batch_size 8 --epochs 50 --workers 8

![image](https://user-images.githubusercontent.com/16435430/87545109-0e1a5b80-c6da-11ea-8c30-bf52d1721de2.png)

**Comment by LCJHust** [comment]
> training using command in tutorial，the utility rate stay at 0 for a very long time，It looks like the training is totally stuck.
> Any suggestion to slove this problem？THX
> 
> > CUDA_VISIBLE_DEVICES=4,5,6,7 OMP_NUM_THREADS=1 python -m torch.distributed.launch --nproc_per_node=4 train.py --launcher pytorch --cfg_file  cfgs/kitti_models/pv_rcnn.yaml --batch_size 8 --epochs 50 --workers 8
> 
> ![image](https://user-images.githubusercontent.com/16435430/87545109-0e1a5b80-c6da-11ea-8c30-bf52d1721de2.png)

I met the same problem with you. Maybe you can change your output path with an empty one. This will help.

**Comment by a43992899** [comment]
I have the same issue, how did you guys fix it? I waited for half an hour and nothing happened, and some of the processes are seemed to be killed.

**Comment by xuanzhangyang** [comment]
I have the same issue, how did you guys fix it?

**Comment by bhyzhao** [comment]
same issue. waiting for the answer.

**Comment by a43992899** [comment]
@sshaoshuai Do you have any idea? Seems like a lot of people encounter this issue.

**Comment by a43992899** [comment]
@sshaoshuai It seems to be a dataloader dead lock? When I set workers<=1, the code can start training in a short time, but when I set it to larger number, this issue will happen. However, I need to use multiple workers so that I can keep my gpu fully utilized. 
BTW, my training machine is aws g4dn.metal, with 2 physical cpus, 24 cores each, and 96 virtual processor, so it's weird that I encounter this issue.

**Comment by MartinHahner** [comment]
@a43992899 Why are you not using AWS EC2 P2 or P3 instances? What is the advantage of G4 instances?

**Comment by a43992899** [comment]
My gpu util rate is unstable, and I find that p3.16xlarge trains slower than g4dn.metal.... both batch_size=16, p3.16xlarge 3~4s/iter, and g4dn.metal 2s+/iter, so I pick the cheaper one.

**Comment by MartinHahner** [comment]
with 8 workers on p3.16xlarge I get 1.73s / iter on average over the entire training
